client/src/store/utils/reducerFunctions.js

```diff
@@ -13,15 +13,58 @@ export const addMessageToStore = (state, payload) => {
 
   return state.map((convo) => {
     if (convo.id === message.conversationId) {
-      convo.messages.push(message);
-      convo.latestMessageText = message.text;
-      return {...convo};
+      const convoCopy = { ...convo };
+      convoCopy.messages.push(message);
+      convoCopy.latestMessageText = message.text;
+      return convoCopy;
```

Comment:
Isn't this change from ticket 1?
Please isolate each PR to its corresponding ticket. You can either rebase this branch to ticket 1 or change base branch of this PR.

Reply:
Yes the change is from fix 1. Not sure why it is showing up here to be honest. The main branch (which contains for Fix #1) is merged on this one. Shouldn't be a problem tho